libs/feature/settings/src/lib/settings-form/student-delete-form.tsx

```diff
@@ -0,0 +1,148 @@
+									title: "Account gelöscht",
+									subtitle: "Sie wurden erfolgreich abgemeldet."
+								});
+							}
+							redirectToLogin();
+						} catch (error) {
+							showToast({
+								type: "error",
+								title: "Account konnte nicht gelöscht werden",
+								subtitle: "Bitte versuchen Sie es erneut."
+							});
+						}
+					}
+					freeDialog("student-delete-form");
+				}}
+			/>,
+			"student-delete-form"
+		);
+	};
+
+	const afterAllDeleteInfoDialog = () => {
+		freeDialog("student-delete-form");
+	};
+
+	return (
+		<div className="mt-8 rounded-lg border border-red-300 bg-red-50 p-6">
+			<h2 className="text-lg font-bold text-red-700">⚠️ Danger Zone</h2>
+			<p className="mt-2 text-sm text-red-600">
+				Sei vorsichtig! Diese Aktionen können nicht rückgängig gemacht werden.
+			</p>
+			<DialogHandler id="student-delete-form" />
+
+			<div className="mt-6 flex flex-col gap-4">
+				<div className="flex items-center gap-2">
+					<button
+						className="btn rounded-full bg-red-500 p-2 text-white hover:bg-red-600 focus:ring-4 focus:ring-red-300"
+						title="Userdaten löschen"
+						onClick={() => {
+							dispatchDialog(
+								<StudentDeleteInfoDialog
+									onClose={() => {
+										freeDialog("student-delete-form");
+										afterPersonalDeleteInfoDialog();
+									}}
+								/>,
+								"student-delete-form"
+							);
+						}}
+					>
+						Userdaten löschen
+					</button>
+					<PersonalDataTooltip />
+				</div>
+
+				<div className="flex items-center gap-2">
+					<button
+						className="btn rounded-full bg-red-500 p-2 text-white hover:bg-red-600 focus:ring-4 focus:ring-red-300"
+						title="Alle Daten löschen"
+						onClick={() => {
+							dispatchDialog(
+								<StudentAllDeleteInfoDialog
+									onClose={() => {
+										freeDialog("student-delete-form");
+										afterAllDeleteInfoDialog();
+									}}
+								/>,
+								"student-delete-form"
+							);
+						}}
+					>
+						Alle Daten löschen
+					</button>
+					<AllDataTooltip />
+				</div>
+			</div>
+		</div>
+	);
+}
+
+const personalDataDeletionInfo = [
+	"profile information",
+	"account settings",
+	"saved preferences",
+	"skill repositorys"
+];
+
+const allDataDeletionInfo = [...personalDataDeletionInfo, "created courses", "created lessons"];
+
+function generateDeletionTooltip(deletionInfo: string[]): string {
+	return `This will delete the following: ${deletionInfo.join(", ")}.`;
+}
+
+export function PersonalDataTooltip() {
+	const personalDataTitle = generateDeletionTooltip(personalDataDeletionInfo);
+
+	return (
+		<Tooltip title={personalDataTitle}>
+			<QuestionMarkTooltip tooltipText="Here you can delete your personal data." />
+		</Tooltip>
+	);
+}
+
+function AllDataTooltip() {
+	const allDataTitle = generateDeletionTooltip(allDataDeletionInfo);
+
+	return (
+		<Tooltip title={allDataTitle}>
+			<QuestionMarkTooltip tooltipText="Here you can delete all your data." />
+		</Tooltip>
+	);
+}
```

Comment:
die struktur ist okay, allerdings für den zweck deutlich zu überladen. Es geht hier nur um das Anzeigen eines Tooltips, ohne inhärente Logik. 

Vorschlag ist 


```

function partialDeleteTooltipDisplayText() {
	return "profile information, account settings, saved preferences, ";
}

function fullDeleteTooltipDisplayText() {
	return `${partialDeleteTooltipDisplayText()} created courses, created lessons, skill repositorys`;
}
```

und dann direkt in der Form:

```
<Tooltip title={partialDeleteTooltipDisplayText}>
    <QuestionMarkTooltip tooltipText="Here you can delete all your data." />
</Tooltip>
......
<Tooltip title={fullDeleteTooltipDisplayText}>
    <QuestionMarkTooltip tooltipText="Here you can delete all your data." />
</Tooltip>
```